cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/task/LoadMonitorTaskRunner.java

```diff
@@ -262,21 +264,29 @@ public void shutdown() {
   /**
    * Pause the scheduled sampling tasks..
    */
-  public void pauseSampling() {
+  public synchronized void pauseSampling() {
     if (_state.get() != PAUSED && !_state.compareAndSet(RUNNING, PAUSED)) {
       throw new IllegalStateException("Cannot pause the load monitor because it is in " + _state.get() + " state.");
     }
+    _awaitExecution = true;
   }
 
   /**
    * Resume the scheduled sampling tasks.
    */
-  public void resumeSampling() {
+  public synchronized void resumeSampling() {
     if (_state.get() != RUNNING && !_state.compareAndSet(PAUSED, RUNNING)) {
       throw new IllegalStateException("Cannot resume the load monitor because it is in " + _state.get() + " state");
     }
+    _awaitExecution = false;
```

Comment:
With the changes we have in `pauseSampling()`, I think we should remove this variable from here.